packages/api/src/dao/EpochDAO.js

```diff
@@ -0,0 +1,43 @@
+const TransactionsDAO = require('../dao/TransactionsDAO')
+const BlocksDAO = require('../dao/BlocksDAO')
+const { calculateEpoch } = require('../utils')
+
+module.exports = class EpochDAO {
+  constructor (knex, genesisTime) {
+    this.blocksDAO = new BlocksDAO(knex)
+    this.transactionsDAO = new TransactionsDAO(knex)
+    this.knex = knex
+    this.genesisTime = genesisTime
+  }
+
+  getEpochInfo = async (index) => {
+    const [currentBlock] = (await this.blocksDAO.getBlocks(1, 1, 'desc')).resultSet
+
+    const epoch = calculateEpoch({
+      index,
+      genesisTime: this.genesisTime,
+      timestamp: currentBlock.header.timestamp
+    })
+
+    if (!(epoch.startTime instanceof Date && !isNaN(epoch.startTime))) {
```

Comment:
I don't think you even need these checks. You always have epoch

Reply:
✅